server/utils/regex.ts

```diff
@@ -0,0 +1,7 @@
+/** This regex searches for YouTube Video IDs from a YouTube URL */
+export const YOUTUBE_VIDEO_ID_REGEX = /(?:youtube\.com\/(?:[^\/]+\/.+\/|(?:v|e(?:mbed)?)\/|.*[?&]v=)|youtu\.be\/)([^"&?\/\s]{11})/;
+
+/** These regexes allow us to find hours, minutes and seconds from a ISO 8601 time string */
```

Comment:
Is it better to parse the ISO8601 string into a date object and use the native methods?

Reply:
I'm not quite sure, could also work since I didn't worked with ISO8601 before. Would you go from the Date.now() if its possible to calc the end-date and get the video duration in numbers from there?

Reply:
Apparently this is a time duration which is different from a Date and JS doesn't have native support for it. I would maybe not store the durationObject in the state though, it's confusing to have both it and duration in seconds there, and possibly not make the getVideoDurationObject helper exported, we should only expose the getVideoDuration() as public and encourage callers to use that.